### PR TITLE
PR to resolve issue [incubator/kafka] Add support for imagePullSecrets #18227

### PR DESCRIPTION
**Pull request by ghost**
Hey,

I've added the condition and stanza needed to define imagePullSecrets to pull the broker image from a private registry. 

I tested the deployment in one of our kubernetes clusters and 
```
...
imagePullSecrets: []
...
```
and 
```
...
imagePullSecrets:
  - name: <secretName>
...
```
worked fine with `helm install kafka/ --name kafka --namespace <test-namespace>`.

### CODE REVIEW

File: incubator/zookeeper/values.yaml [hunk 1]
```diff
@@ -16,10 +16,14 @@ updateStrategy:
 ## - https://github.com/kubernetes/contrib/tree/master/statefulsets/zookeeper
 ## - https://github.com/kubernetes/contrib/blob/master/statefulsets/zookeeper/Makefile#L1
 image:
-  repository: zookeeper     # Container image repository for zookeeper container.
+  repository: dtr.rz.sys.aok.de/occonnect/zookeeper     # Container image repository for zookeeper container.
```
Comment (benjigoldberg) [inline incubator/zookeeper/values.yaml]: can you change this back?

File: incubator/zookeeper/values.yaml [hunk 2]
```diff
@@ -16,10 +16,14 @@ updateStrategy:
...
   tag: 3.5.5                # Container image tag for zookeeper container.
   pullPolicy: IfNotPresent  # Image pull criteria for zookeeper container.
 
+## Specify a secret with credentials to pull the image from a private registry
+imagePullSecrets: 
+  - name: registry-credentials
```
Comment (benjigoldberg) [inline incubator/zookeeper/values.yaml]: dont want to check this in

File: incubator/zookeeper/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: zookeeper
 home: https://zookeeper.apache.org/
-version: 2.1.0
+version: 2.1.1
```
Comment (benjigoldberg) [inline incubator/zookeeper/Chart.yaml]: lets bump minor instead of patch

**Review by benjigoldberg — CHANGES_REQUESTED** [review]

File: incubator/zookeeper/values.yaml (hunk 1 above)
Comment (ghost) [reply incubator/zookeeper/values.yaml]: Whoops, sorry! Removed it.

File: incubator/zookeeper/values.yaml (hunk 2 above)
Comment (ghost) [reply incubator/zookeeper/values.yaml]: True, thank you!

**Comment by zanhsieh** [comment]
@OlGe93 
Kafka needs a version bumped to latest.

**Comment by OlGe93** [comment]
@zanhsieh 
Where should the version be bumped and why to latest? :)

**Comment by zanhsieh** [comment]
@OlGe93 
Check the ci/circleci: lint-charts below:
```
==> Linting incubator/zookeeper
Lint OK

1 chart(s) linted, no failures
Error linting charts: Error processing charts
------------------------------------------------------------------------------------------------------------------------
 ✖︎ kafka => (version: "0.20.1", path: "incubator/kafka") > Chart version not ok. Needs a version bump!
 ✔︎ zookeeper => (version: "2.2.0", path: "incubator/zookeeper")
------------------------------------------------------------------------------------------------------------------------
Exited with code 1
```

**Pull request closed without merging**